nomad/operator_endpoint_test.go

```diff
@@ -368,6 +375,209 @@ func TestOperator_RaftRemovePeerByID_ACL(t *testing.T) {
+			c.BootstrapExpect = cSize
+			c.ACLEnabled = args.enableACL
+		})
+	}
+	t.Cleanup(out.Cleanup)
+	out.rpc = out.server[0].RPC
+
+	TestJoin(t, out.server...)
+	out.WaitForLeader()
+
+	if args.enableACL {
+		// Bootstrap the ACL subsystem
+		token := mock.ACLManagementToken()
+		err := out.server[0].State().BootstrapACLTokens(structs.MsgTypeTestSetup, 1, 0, token)
+		if err != nil {
+			t.Fatalf("failed to bootstrap ACL token: %v", err)
+		}
+		t.Logf("bootstrap token: %v", *token)
+		out.token = token
+	}
+	return out
+}
+
+func (tc testcluster) WaitForLeader() {
+	testutil.WaitForLeader(tc.t, tc.rpc)
+}
+
+func (tc testcluster) leader() *Server {
+	tc.WaitForLeader()
+	for _, s := range tc.server {
+		if isLeader, _ := s.getLeader(); isLeader {
+			return s
+		}
+	}
+	return nil
+}
+
+func (tc testcluster) anyFollower() *Server {
+	if len(tc.server) < 2 {
+		return nil
+	}
+
+	testutil.WaitForLeader(tc.t, tc.rpc)
+	for _, s := range tc.server {
+		if isLeader, _ := s.getLeader(); !isLeader {
+			return s
+		}
+	}
+	// something weird happened.
+	return nil
+}
+
+func TestOperator_TransferLeadershipToServerAddress_ACL(t *testing.T) {
+	ci.Parallel(t)
+
+	tc := newTestCluster(t, tcArgs{enableACL: true})
+	s1 := tc.leader()
+	codec := rpcClient(t, s1)
+	state := s1.fsm.State()
+
+	lAddr, _ := s1.raft.LeaderWithID()
+
+	var addr raft.ServerAddress
+	// Find the first non-leader server in the list.
+	for a := range s1.localPeers {
+		addr = a
+		if addr != lAddr {
+			break
+		}
+	}
+
+	// Create ACL token
+	invalidToken := mock.CreatePolicyAndToken(t, state, 1001, "test-invalid", mock.NodePolicy(acl.PolicyWrite))
+
+	arg := structs.RaftPeerRequest{
+		Address:      addr,
+		WriteRequest: structs.WriteRequest{Region: s1.config.Region},
+	}
+
+	var reply struct{}
+
+	t.Run("no-token", func(t *testing.T) {
+		// Try with no token and expect permission denied
+		err := msgpackrpc.CallWithCodec(codec, "Operator.TransferLeadershipToPeer", &arg, &reply)
+		must.Error(t, err)
+		must.ErrorIs(t, err, rpcPermDeniedErr)
+	})
+
+	t.Run("invalid-token", func(t *testing.T) {
+		// Try with an invalid token and expect permission denied
+		arg.AuthToken = invalidToken.SecretID
+		err := msgpackrpc.CallWithCodec(codec, "Operator.TransferLeadershipToPeer", &arg, &reply)
+		must.Error(t, err)
+		must.ErrorIs(t, err, rpcPermDeniedErr)
+	})
+
+	t.Run("good-token", func(t *testing.T) {
+		// Try with a management token
+		arg.AuthToken = tc.token.SecretID
+		err := msgpackrpc.CallWithCodec(codec, "Operator.TransferLeadershipToPeer", &arg, &reply)
+		must.NoError(t, err)
+
+		// Is the expected leader the new one?
+		tc.WaitForLeader()
+		lAddrNew, _ := s1.raft.LeaderWithID()
+		must.Eq(t, addr, lAddrNew)
+	})
+}
+
+func TestOperator_TransferLeadershipToServerID_ACL(t *testing.T) {
+	ci.Parallel(t)
+	tc := newTestCluster(t, tcArgs{enableACL: true})
+	s1 := tc.leader()
+	codec := rpcClient(t, s1)
+	state := s1.fsm.State()
+
+	_, ldrID := s1.raft.LeaderWithID()
```

Comment:
Nitpick: setting up the test cluster is so expensive relative to the test cases, that it might make sense to stand up one test cluster for both `TestOperator_TransferLeadershipToServerID_ACL` and `TestOperator_TransferLeadershipToServerAddress_ACL` and then just send different requests for the only two cases that actually care about that difference. This will help avoid making test run times worse.

Reply:
We can do this in a separate PR, if we feel like it later. Test setup is kinda rough.